pkg/proxy/handlers/spacelister_get.go

```diff
@@ -26,6 +28,8 @@ func HandleSpaceGetRequest(spaceLister *SpaceLister, GetMembersFunc cluster.GetM
 	// get specific workspace
 	return func(ctx echo.Context) error {
 		requestReceivedTime := ctx.Get(regsercontext.RequestReceivedTime).(time.Time)
+		publicViewerEnabled := configuration.GetRegistrationServiceConfig().PublicViewerEnabled()
+		ctx.Set(context.PublicViewerEnabled, publicViewerEnabled)
```

Comment:
Instead of explicitly set the context in every affected handler I would add another middleware which would set the public viewer flag for the context.
This is where we register middlewares for the proxy: https://github.com/codeready-toolchain/registration-service/blob/007cba61412e49882a3055eff63c9a18081f63d4/pkg/proxy/proxy.go#L106

Reply:
good idea, I removed it in 9f455b3d03b96c6488bf51bdc701d1b93b87e237 and I will create a new PR with the middleware

Reply:
PR created: https://github.com/codeready-toolchain/registration-service/pull/453